airbyte-integrations/connectors/source-dixa/source_dixa/source.py

```diff
@@ -87,83 +77,59 @@ def request_params(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> Mu
+        return {self.cursor_field: max(current_stream_state.get(self.cursor_field, 0), latest_record.get(self.cursor_field, 0))}
 
     def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs):
         """
-        Returns slices of size self.batch_size.
+        Returns slices of size self.batch_size
         """
         slices = []
 
         stream_state = stream_state or {}
         # If stream_state contains the cursor field and the value of the cursor
         # field is higher than start_timestamp, then start at the cursor field
         # value. Otherwise, start at start_timestamp.
-        updated_after = max(stream_state.get(ConversationExport.cursor_field, 0), self.start_timestamp)
+        updated_after = max(stream_state.get(self.cursor_field, 0), self.start_timestamp)
         while updated_after < self.end_timestamp:
-            updated_before = min(
-                ConversationExport.add_days_to_ms_timestamp(days=self.batch_size, milliseconds=updated_after), self.end_timestamp
-            )
+            updated_before = min(utils.add_days_to_ms_timestamp(days=self.batch_size, milliseconds=updated_after), self.end_timestamp)
             slices.append({"updated_after": updated_after, "updated_before": updated_before})
             updated_after = updated_before
         return slices
 
+
+class ConversationExport(IncrementalDixaStream):
     def path(self, **kwargs) -> str:
         return "conversation_export"
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
-        """
-        Uses the `updated_at` field, which is a Unix timestamp with millisecond precision.
-        """
-        if current_stream_state is not None and ConversationExport.cursor_field in current_stream_state:
-            return {
-                ConversationExport.cursor_field: max(
-                    current_stream_state[ConversationExport.cursor_field], latest_record[ConversationExport.cursor_field]
-                )
-            }
-        else:
-            return {ConversationExport.cursor_field: self.start_timestamp}
-
 
 class SourceDixa(AbstractSource):
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         """
-        Try loading one day's worth of data.
+        Check connectivity using one day's worth of data.
         """
+        auth = TokenAuthenticator(token=config["api_token"]).get_auth_header()
+        url = "https://exports.dixa.io/v1/conversation_export"
+        start_date = datetime.strptime(config["start_date"], "%Y-%m-%d")
+        start_timestamp = utils.datetime_to_ms_timestamp(start_date)
+        params = {
+            "updated_after": start_timestamp,
+            "updated_before": utils.add_days_to_ms_timestamp(days=1, milliseconds=start_timestamp),
+        }
+
```

Comment:
Reuse some stream for this.
This code part duplicates the Dixa logic here and it will be able to  complicate its support in the future.

Reply:
Done.